crates/sui-indexer-alt-jsonrpc/src/data/kv_loader.rs

```diff
@@ -116,4 +130,107 @@ impl KvLoader {
                 .transpose(),
         }
     }
+
+    pub(crate) async fn load_one_transaction(
+        &self,
+        digest: TransactionDigest,
+    ) -> Result<Option<TransactionContents>, Arc<Error>> {
+        let key = TransactionKey(digest);
+        match self {
+            Self::Bigtable(loader) => Ok(loader
+                .load_one(key)
+                .await?
+                .map(TransactionContents::Bigtable)),
+            Self::Pg(loader) => Ok(loader.load_one(key).await?.map(TransactionContents::Pg)),
+        }
+    }
+}
+
+impl TransactionContents {
+    pub(crate) fn data(&self) -> anyhow::Result<TransactionData> {
+        match self {
+            Self::Pg(stored) => {
+                let data: TransactionData =
+                    bcs::from_bytes(&stored.raw_transaction).map_err(|e| {
+                        anyhow::anyhow!("Failed to deserialize transaction data: {}", e)
+                    })?;
```

Comment:
Use `context` for this.

```suggestion
                    bcs::from_bytes(&stored.raw_transaction)
                        .context("Failed to deserialize transaction data");
```